### ISSUE TITLE
Model Weights for HRSC 2016 dataset?

### ISSUE THREAD

**Issue by Sairam13001** [zhaozeyang108/Oriented-DETR#9]
Hi @zhaozeyang108, thank you for making the code public. Do you plan to make the configs and model weights public for HRSC2016 dataset in the recent future?

**Comment by zhaozeyang108** [comment]
> Hi @zhaozeyang108, thank you for making the code public. Do you plan to make the configs and model weights public for HRSC2016 dataset in the recent future?

I apologize for the inconvenience,but the server for this portion of the experiment has unfortunately encountered a failure,resulting in the loss of our weights.I am currently unable to allocate time to rerun the experiment,so I would greatly appreciate it if you could take on this task.The hyperparameters should be set in the same manner as they were for the DIOR dataset.The details regarding the epoch settings and the learning rate adjustment strategies are documented in the paper.For the pre-trained weights,you may choose to use those from either the DOTA or DIOR datasets.If you encounter any issues during the reproduction process, feel free to ask questions.

**Comment by Sairam13001** [comment]
Hi @zhaozeyang108,

I tried reproducing the results on HRSC2016 dataset. I used DIOR dataset's pre-trained model weights for initialisation and trained for 50 epochs with learning rate as 0.0001, weight decay as 0.0001 and learning rate drop at epochs 33 and 45 as mentioned in the paper. However, the mAP is coming out to be very less than expected. I am attahcing a .txt file that shows the results.

[r50_hrsc2016.txt](https://github.com/user-attachments/files/18284685/r50_hrsc2016.txt)

Am I doing something wrong? Please let me know.

**Comment by zhaozeyang108** [comment]
> Hi @zhaozeyang108,
> 
> I tried reproducing the results on HRSC2016 dataset. I used DIOR dataset's pre-trained model weights for initialisation and trained for 50 epochs with learning rate as 0.0001, weight decay as 0.0001 and learning rate drop at epochs 33 and 45 as mentioned in the paper. However, the mAP is coming out to be very less than expected. I am attahcing a .txt file that shows the results.
> 
> [r50_hrsc2016.txt](https://github.com/user-attachments/files/18284685/r50_hrsc2016.txt)
> 
> Am I doing something wrong? Please let me know.

Is there a mistake in the number of categories you've set? Isn't HRSC supposed to be just one category? Why did I see results testing for two categories?

**Comment by Sairam13001** [comment]
So, in the config file, should I set the num_classes to 1?

**Comment by zhaozeyang108** [comment]
> So, in the config file, should I set the num_classes to 1?

The number of categories in the config should be set to 2 (category count 1 + background). The calculation in the evaluate code should be unrelated to the category settings in the config and should be related to the JSON file. You can debug the evaluate section of the code to check if the map calculation is ultimately divided by 2. If it is, that's the issue. You can also visualize to check if the detection is correct (set the confidence level to 0.15 in the config).

**Comment by Sairam13001** [comment]
Okay, understand. Thank you for the clarity.

**Comment by lwh-Labs** [comment]
> Okay, understand. Thank you for the clarity.好的，明白了。谢谢你的澄清。

Hello, I would like to ask if the MAP results for your HSRC dataset ended up being the same as in the paper?

**Comment by gsenxu** [comment]
<img width="759" height="462" alt="Image" src="https://github.com/user-attachments/assets/ca215e4b-b16f-414a-928d-bcd07453a41b" />
Great work！What do these weights represent?

**Comment by gsenxu** [comment]
Why the  num_classes matching swin_t_dota_ss.pth is set to 91 instead of 16?
